tests/pkg_tiny-asn1/main.c

```diff
@@ -0,0 +1,207 @@
+ * directory for more details.
+ */
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <inttypes.h>
+
+#include "tiny-asn1.h"
+
+void print_hex(uint8_t *data, unsigned int len)
+{
+    unsigned int count = 0;
+    unsigned int blockCount = 0;
+
+    while (count < len) {
+        printf("%02x ", data[count]);
+        ++count;
+        ++blockCount;
+        if (blockCount == 4) {
+            printf("  ");
+        }
+        if (blockCount == 8) {
+            printf("\n");
+            blockCount = 0;
+        }
+    }
+    printf("\n");
+}
+
+
+void print_asn1(asn1_tree *list, int depth)
+{
+    printf("d=%d, Tag: %02x, len=%" PRIu32 "\n", depth, list->type, list->length);
+    if (list->child == NULL) {
+        printf("Value:\n");
+        print_hex(list->data, list->length);
+    }
+    else {
+        print_asn1(list->child, depth + 1);
+    }
+
+    if (list->next != NULL) {
+        print_asn1(list->next, depth);
+    }
+}
+
+
+
+int main(void)
+{
+    /* Created with echo "12345678901234567890"|openssl cms -EncryptedData_encrypt -aes128 -secretkey 39904F36D98779D00F9A2B8139D2957F -outform der|xxd --i */
+    uint8_t cms_data[] = {
+        0x30, 0x60, 0x06, 0x09, 0x2a, 0x86, 0x48, 0x86, 0xf7, 0x0d, 0x01, 0x07,
+        0x06, 0xa0, 0x53, 0x30, 0x51, 0x02, 0x01, 0x00, 0x30, 0x4c, 0x06, 0x09,
+        0x2a, 0x86, 0x48, 0x86, 0xf7, 0x0d, 0x01, 0x07, 0x01, 0x30, 0x1d, 0x06,
+        0x09, 0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x01, 0x02, 0x04, 0x10,
+        0x4d, 0xaa, 0x7b, 0x74, 0xfb, 0xf7, 0x7a, 0xfa, 0xb0, 0x23, 0xa2, 0x9c,
+        0x1a, 0xb6, 0x51, 0xe4, 0x80, 0x20, 0x6b, 0x8f, 0x64, 0xd7, 0x3a, 0x81,
+        0x01, 0x0c, 0x7b, 0xab, 0xfd, 0x26, 0x6d, 0x46, 0x2e, 0x3e, 0xc4, 0xf6,
+        0x01, 0x78, 0x8a, 0xc0, 0xc0, 0x52, 0x0a, 0x66, 0x17, 0x71, 0x4f, 0xf6,
+        0xd9, 0x95
+    };
+
+    int32_t asn1_object_count = der_object_count(cms_data, sizeof(cms_data));
+
+    if (asn1_object_count < 0) {
+        fprintf(stderr, "ERROR: Could not calculate the number of Elements within the data.\n");
+        return 1;
+    }
+
+    asn1_tree *asn1_objects = (asn1_tree *)(malloc(sizeof(asn1_tree) * asn1_object_count));
+    if (asn1_objects == NULL) {
+        fprintf(stderr, "ERROR: Could not allocate the memory for the ASN.1 objects.\n");
+        return 1;
+    }
+
+    asn1_tree cms;
+
+    if (der_decode(cms_data, sizeof(cms_data), &cms, asn1_objects, asn1_object_count) < 0) {
+        fprintf(stderr, "ERROR: Could not parse the data.\n");
+        return 1;
+    }
+
+    /* Dump the data */
+    print_asn1(&cms, 0);
+
+    /* Since we know this is CMS data, we can try to interpret it. */
+    if (cms.type != 0x30) {
+        fprintf(stderr, "ERROR: The outer type is not a SEQUENCE.\n");
+        return 1;
+    }
+
+    asn1_tree *content_type = cms.child;
+    if (content_type == NULL || content_type->type != 0x06) {
```

Comment:
Parenthesis are round braces ;) I mean e. g. 

```C
if ((content_type == NULL) || (content_type->type != 0x06)) {
```

instead of

```C
if (content_type == NULL || content_type->type != 0x06) {
```

See https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions#indentation-and-braces